django/db/models/sql/query.py

```diff
@@ -1918,6 +1918,32 @@ def clear_ordering(self, force_empty):
         if force_empty:
             self.default_ordering = False
 
+    def _get_alias_map_tables_column_names(self):
+        def get_cols(model):
+            cols = []
+            for field in model._meta.get_fields():
+                if hasattr(field, 'column'):
+                    cols.append(field.column)
+            return cols
+
+        tables = {self.model._meta.db_table: self.model}
+        cols = set(get_cols(self.model))
```

Comment:
It would likely be easier to build a set of models from `self.model` and then iterate over all `Join` entries in `self.alias_map` and retrieve the model from `join_field.model`?

I haven't tested it but something along these lines should work. By the way you can exclude `self.model` since name collisions are already avoided at `annotate` time, it's only joined tables that cause issues.

```python
joined_models = {}
for join in list(self.alias_map.values())[1:]:
    joined_models.add(join.join_field.model)
return set(chain.from_iterable(
    (field.column for field in model._meta.local_concrete_fields) for model in joined_models
))
```

Reply:
Thanks @charettes for your suggested code.
I've added `join.join_field.related_model` to `joined_models` as well. otherwise, the added test failed.